core/integration/module_test.go

```diff
@@ -41,6 +41,110 @@ func TestModule(t *testing.T) {
 	testctx.Run(testCtx, t, ModuleSuite{}, Middleware()...)
 }
 
+func (ModuleSuite) TestModuleNamingCompatArgName(ctx context.Context, t *testctx.T) {
+	for _, tc := range []string{"new", "old"} {
+		c := connect(ctx, t)
+
+		devModGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--source=.", "--name=minimal", "--sdk=go")).
+			WithNewFile("main.go", `package main
+	
+	import (
+		"dagger/minimal/internal/dagger"
+	)
+	
+	type Minimal struct {
+		Config dagger.JSON
+	}
+	
+	func New() *Minimal {
+		return &Minimal{
+			Config: "{\"a\":1}",
+		}
+	}
+
+	func (m *Minimal) WithSecondFunction(skipTParse string) *dagger.Container {
+		return dag.Container().From("alpine:latest").WithExec([]string{"echo", skipTParse})
+	}
+	`,
+			)
+
+		if tc == "new" {
+			out, err := devModGen.
+				With(daggerQuery(`{minimal{withSecondFunction(skipTParse:"hello"){stdout}}}`)).
+				Stdout(ctx)
+			require.NoError(t, err)
+			require.JSONEq(t, `{"minimal":{"withSecondFunction":{"stdout":"hello\n"}}}`, out)
+		} else {
+			// now simulate old version
+			versionAModGen := devModGen.WithNewFile("dagger.json", `{
+			  "name": "minimal",
+			  "sdk": "go",
+			  "engineVersion": "v0.12.5"
+			}`)
+			out, err := versionAModGen.
+				With(daggerQuery(`{minimal{withSecondFunction(skipTparse:"hello"){stdout}}}`)).
+				Stdout(ctx)
+			require.NoError(t, err)
+			require.JSONEq(t, `{"minimal":{"withSecondFunction":{"stdout":"hello\n"}}}`, out)
+		}
+	}
+}
+
+func (ModuleSuite) TestModuleNamingCompatFuncName(ctx context.Context, t *testctx.T) {
```

Comment:
Hm, I suspect there's still something missing here :cry: 

Suppose:
- We have a dep using "new dagger" that was one of these weird cases like `WithDaggerCLIAlpine`
- We have a main module that's using an "old dagger" that install `WithDaggerCLIAlpine`
- What does the codegen look like for the main module? I think it probably needs to generate with the "old dagger" form.

Maybe this "just works" as is, but I'd like a test to confirm that (and ideally for go/python/typescript consumers).

If this is an issue, then the fix will be to propagate the `strcase.Caser` code into `cmd/codegen` and each of the runtime modules. Thankfully we already propagate the `engineVersion` into there, so *hopefully* wouldn't be too bad with all the context plumbing you've already done :tada: